### PR TITLE
Experimental options to analyze and tag compatibility with a future SDK.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: lib/src/package_analyzer.dart [hunk 1]
```diff
@@ -39,6 +39,10 @@ class InspectOptions {
   /// Whether pana should also access the remote repository specified in pubspec.yaml.
   final bool checkRemoteRepository;
 
+  /// If specified, pana will analyze the package with future SDKs configured in
+  /// [ToolEnvironment], and grant this tag if the analysis passes without errors.
+  final String? futureSdkTag;
```
Comment (jonasfj) [inline lib/src/package_analyzer.dart:44]: This is a path to a future SDK, right?
```suggestion
  final String? futureSdk;
```

**Review by jonasfj — APPROVED** [review]

File: lib/src/package_analyzer.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (isoos) [reply lib/src/package_analyzer.dart:44]: We don't need to, however, this is a convenient way to trigger this part of the analysis, without having another SDK to be downloaded and configured. Without this, we would need either a new flag here, or a properly configured future SDK in the `ToolEnvironment`, which is a bit of a hassle in the tests.

File: lib/src/package_analyzer.dart [hunk 2]
```diff
@@ -193,6 +200,13 @@ class PackageAnalyzer {
         // tags are exposed, explanations are ignored
         // TODO: use a single result object to derive tags + report
         tags.addAll(tags_);
+
+        // When analysis was successful up until this point, also analyse with
+        // future SDKs and award tag - if available.
+        if (options.futureSdkTag != null &&
+            await context.isCompatibleWithFutureSdk) {
+          tags.add(options.futureSdkTag!);
+        }
```
[the remark this answers is not in the document]
Comment (isoos) [reply lib/src/package_analyzer.dart:209]: I think that we should use `is:dart-3-compatible` only when we mean it (and we run it on the Dart 3 SDK). Until then we can use just a vague `is:future-sdk-compatible` tag.

[the remark this answers is not in the document]
Comment (jonasfj) [reply lib/src/package_analyzer.dart:209]: okay

File: lib/src/package_analyzer.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonasfj) [reply lib/src/package_analyzer.dart:44]: okay..

**Pull request merged**